crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -156,6 +156,22 @@ def compute_resources_available?
         true
       end
 
+      # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+      def ha_presence_check
```

Comment:
@rsalevsky I think the relation of these kind of methods is addressed in https://github.com/crowbar/crowbar-core/pull/672. It would be probably best if we could merge that one first and adapt the rest later